inspire_disambiguation/api.py

```diff
@@ -127,7 +148,11 @@ def cluster(ethnicity_model_path, distance_model_path, n_jobs, signature_block=N
             clustered. If set to None, clustering will run on all blocks.
     """
     start_time = datetime.now()
-    signatures = get_signatures(signature_block=signature_block)
+    # validation signatures (the same as in the distance model)
+    if validation_signatures:
+        signatures = validation_signatures
```

Comment:
this is wrong, you need to cluster on the whole block, but use these signatures for scoring (as in the [BEARD example](https://github.com/inspirehep/beard/blob/8d21e6757ac90a6e704f5797fc6cecc784563b22/examples/applications/author-disambiguation/clustering.py#L242-L265)), so I think you'll need to make changes inside the clustering, not simply at the API level.